FLASK/todolist_api/task.py

```diff
@@ -1,39 +1,54 @@
-
-
-class Task(Resource):
-
-    @classmethod
-    def get(cls, task_id):
-        if task_id == "all":
-            return tasks
-        if task_id not in tasks:
-            abort(404, message=f'Task {task_id} was not found!')
-        return tasks[task_id]
-
-    @classmethod
-    def put(cls, task_id):
-        args = parser.parse_args()
-        new_task = {'name': args['name'],
-                    'dueDate': args['dueDate']}
-
-        if not new_task['dueDate']:
-            new_task['dueDate'] = datetime.today().year
-
-        tasks[task_id] = new_task
-        write_changes_to_file()
-        return {task_id: tasks[task_id]}, 201
-
-    @classmethod
-    def delete(cls, task_id):
-        if task_id not in tasks:
-            abort(404, message=f'Task {task_id} was not found!')
-        del tasks[task_id]
-        write_changes_to_file()
-        return "", 204
+from typing import List, Optional
+
+
+class Task:
+
+    def __init__(self,
+                 name: str,
+                 assigner: str,
+                 company: str,
+                 deadline: Optional[datetime] = None,
+                 priority: Optional[int] = 0,
+                 description: Optional[str] = "",
+                 status: str = "Not started",
+                 assigned_personnel: Optional[List[str]] = None,
+                 creation_date: Optional[datetime] = None,
+                 last_modified_date: Optional[datetime] = None,
+                 comments: Optional[str] = ""
+                 ):
+
+        self.name = name
+        self.assigner = assigner
+        self.company = company
+        self.deadline = deadline
+        self.priority = priority if priority in [1, 2, 3] else 0
+        self.description = description
+        self.status = status
+        self.assigned_personnel = assigned_personnel
+        self.creation_date = creation_date if creation_date else datetime.now()
+        self.last_modified_date = last_modified_date if last_modified_date else self.creation_date
+        self.comments = comments
+
+    # Comparison methods
+    def __eq__(self, other):
+        return self.priority == other.priority
+
+    def __lt__(self, other):
+        return self.priority < other.priority
+
+    def is_overdue(self):
+        if self.deadline < datetime.now():
+            return f"Task is overdue!"
+        elif self.deadline >= datetime.now():
+            remaining_time = self.deadline - datetime.now()
+            return f"You still have {remaining_time.days} days to do the task!"
+
+    def add_comment(self, new_comment):
+        self.comments += f"\n {new_comment}"
+        self.last_modified_date = datetime.now()
+
+    def __repr__(self):
+        return (f"{self.__class__.__name__}('{self.name}', '{self.assigner}', '{self.company}', "
+                f"{self.priority}, '{self.description}', '{self.status}', {self.assigned_personnel}, "
+                f"'{self.creation_date}', '{self.last_modified_date}', '{self.comments}')")
```

Comment:
```python
'{self.comments}'
```
Din ce vad, se pare ca aici nu iti returneaza o liste de comentarii, oare daca ai face un for each pe fiecare comment si sa il adaugi intr-un string, iar apoi sa il pui in lista nu ar fi mai usor?